internal/controller/manager.go

```diff
@@ -200,6 +200,20 @@ func StartManager(cfg config.Config) error {
 		return fmt.Errorf("cannot register grpc server: %w", err)
 	}
 
+	agentLabelCollector := telemetry.NewLabelCollectorConfigImpl(telemetry.LabelCollectorConfig{
+		K8sClientReader: mgr.GetAPIReader(),
+		Version:         cfg.GatewayPodConfig.Version,
+		PodNSName: types.NamespacedName{
+			Namespace: cfg.GatewayPodConfig.Namespace,
+			Name:      cfg.GatewayPodConfig.Name,
+		},
+	})
+
+	agentLabels, err := agentLabelCollector.Collect(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to collect agent labels: %w", err)
+	}
+
```

Comment:
Does this need to be done here? Can we just gather this info in the Provisioner's constructor? This manager file is already pretty unwieldy.